### ISSUE TITLE
Filament custom fields available on spools

### ISSUE THREAD

**Issue by cnf** [Donkie/Spoolman#334]
**Is your feature request related to a problem? Please describe.**
I have fields such as "RAL" and "glitter" on filaments, but those also are applicable on the spools of said filament, of course.

**Describe the solution you'd like**
Have the Filament custom fields available / viewable / filterable on the spools view

**Describe alternatives you've considered**
I don't know? I think filament information is relevant on a spool as well?

**Additional context**
None.

Thanks!

**Comment by FBrand** [comment]
Second that.
It would be useful to be able to promote Custom Fields from Vendor to Filament and from Filament to Spool.
It would probably also be useful to add a checkbox in the creation dialogue to to enable/disable the created field being promoted.

**Comment by cnf** [comment]
> are you looking for these to be read-only fields or actual fields on the spool?
> 
> Generally, as I have been digging into the code, the `filament` object is properties of the filament itself, while the spool has some of it's own properties (weight, empty weight, etc). I just proposed a change to allow the empty spool weight (tare weight) to "fall" from a field on vendor to filament to spool, but really, filament shouldn't know (or care) about spool weight of any kind. I left it there to not break anyone's current implementation.
> 
> What it seems you are asking for isn't so much adding new properties to the spool, but just allowing the filament's properties to be "searchable" from the spool endpoint. If that's what you are asking for, I would agree, that makes some sense, but I might implement it as a new endpoint to avoid performance issues with the current endpoints.
> 
> If you are asking for these properties to exist and become defaults from vendor to filament to spool, well, that's a LOT of data duplication and mostly unnecessary.

sorry for the late reply, yes, i want the information to be visible/searchable on a spool, not copied.

If i add a comment on a filament "hard to print" for example, I want to see that on a spool as well.
Or if I add a RAL field, or a "glitter" field, i want it to be shown, read only, on a spool as well.

the instance should have access to the parents properties, really.

Thanks!

**Comment by Donkie** [comment]
Makes sense. In general though I really want to update the table layout before adding something like this. Today it's very bulky, and it's not really possible to make columns manually resizable.

Fixing that would mean finding a new table drawing library though, which is a big change..

**Comment by Donkie** [comment]
This is available in the new client in 0.26.0, try it out!
